pinot-core/src/main/java/org/apache/pinot/core/query/request/context/DataSourceContext.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.query.request.context;
+
+public class DataSourceContext {
+  private final String _tableName;
+  private final QueryContext _subquery;
+  private final JoinContext _join;
```

Comment:
Can you clarify if in this PR, the goal for this is to only capture the local JOIN support and it will evolve later because I see subquery as well and I wonder if that is enough to capture generic subquery support ?

Reply:
Also IIUC - 2 and 3 will be NULL for existing single table Pinot queries. Correct ?